Branch-SDK/Branch-SDK/BNCServerInterface.m

```diff
@@ -191,8 +191,10 @@ - (BNCServerResponse *)genericHTTPRequest:(NSURLRequest *)request log:(BOOL)log
 #pragma mark - Internals
 
 - (NSURLRequest *)prepareGetRequest:(NSDictionary *)params url:(NSString *)url key:(NSString *)key retryNumber:(NSInteger)retryNumber log:(BOOL)log {
-    NSDictionary *preparedParams = [self prepareParamDict:params key:key retryNumber:retryNumber];
+    NSMutableDictionary *preparedParams = [[self prepareParamDict:params key:key retryNumber:retryNumber] mutableCopy];
     
+    // explicitly remove instrumentation info from GET requests. They are only meant to be sent in POST requests.
+    [preparedParams removeObjectForKey:BRANCH_REQUEST_KEY_INSTRUMENTATION];
```

Comment:
I'd prefer that you add this check in the below method, alongside this conditional `if (self.preferenceHelper.instrumentationDictionary.count) {`. I remember we had a history of concurrent modification issues with this type of stuff, which is where the structure of the `prepareParamDict` method came from.